src/swarm.js

```diff
@@ -175,6 +180,7 @@ function Swarm (peerInfo) {
       // if protocol is selected, multistream that protocol
       if (!conn) {
         callback(new Error('Unable to open a connection'))
+        return
```

Comment:
I couldn't easily find an scenario when this can happen but without this `return` you can end up with double callbacks :scream: